### PR TITLE
`sleep`: Replace uucore::parse_time::from_str with fundu

### PR DESCRIPTION
**Pull request by Joining7943**
See also https://github.com/Joining7943/fundu. This pr also improves the error messages.

This pr depends on #4439

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
Marking this as draft because of the dependency between the two PRs. You can mark it ready again when the other PR is merged :)

File: src/uu/sleep/src/sleep.rs [hunk 1]
```diff
@@ -61,14 +62,35 @@ pub fn uu_app() -> Command {
...
                 arg_error = true;
-                show_error!("invalid time interval '{input}'");
+
+                let reason = match error {
+                    ParseError::Empty if input.is_empty() => "Input was empty".to_string(),
+                    ParseError::Empty => "Found only whitespace in input".to_string(),
+                    ParseError::Syntax(pos, description)
+                    | ParseError::TimeUnit(pos, description) => format!(
+                        "{description} at position {}",
+                        pos.checked_add(1).unwrap_or(pos)
```
Comment (tertsdiepraam) [inline src/uu/sleep/src/sleep.rs:82]: Nit: This is just a `saturating_add` isn't it?

**Review by tertsdiepraam — COMMENTED** [review]
Nice! I'll update the branch to get rid of the clippy warnings.

File: src/uu/sleep/src/sleep.rs (hunk 1 above)
Comment (Joining7943) [reply src/uu/sleep/src/sleep.rs:82]: oh yes, I'll change it to `saturating_add`!

> Nice! I'll update the branch to get rid of the clippy warnings.

Thanks!

Comment (Joining7943) [reply src/uu/sleep/src/sleep.rs]: I rebased onto main after this change, so the merge commit is gone again.

**Comment by tertsdiepraam** [comment]
Nice!

**Pull request merged**